### ISSUE TITLE
Build error on Ubuntu 18.04

### ISSUE THREAD

**Issue by Sviluppo718** [mimblewimble/grin#3100]
**Describe the bug**
Hi, I'm getting the following four errors when trying to build Grin 2.1.0 (version 2.0.0 seemed to be functional): 

```
Compiling grin_pool v2.0.0 (/home/myUserName/grin/pool)
error: expected #[serde(...)]
  --> pool/src/types.rs:94:4
   |
94 |     #[serde = "default_accept_fee_base"]
   |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: expected #[serde(...)]
  --> pool/src/types.rs:98:4
   |
98 |     #[serde = "default_max_pool_size"]
   |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: expected #[serde(...)]
   --> pool/src/types.rs:102:4
    |
102 |     #[serde = "default_max_stempool_size"]
    |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: expected #[serde(...)]
   --> pool/src/types.rs:108:4
    |
108 |     #[serde = "default_mineable_max_weight"]
    |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 4 previous errors

error: Could not compile `grin_pool`.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

My ``` $ cargo --version```
outputs: 
cargo 1.38.0 (23ef9a4ef 2019-08-20)

```$ rustup --version```
outputs: rustup 1.20.2 (13979c968 2019-10-16) 

**To Reproduce**
```cargo build --release``` in the grin (top-level) directory

**Desktop (please complete the following information):**
```$ lsb_release -a```
outputs [...] Description:  Ubuntu 18.04.3 LTS

**Additional context**
Alas I've got really limited expertise with rust (at this point).  Eventually I'd like to remedy this.

**Comment by quentinlesceller** [comment]
Can you checkout the latest commit on branch `milestone/v2.1.x`. That should do.

**Comment by Sviluppo718** [comment]
Sorry -- in the context I provided it should have included "**git newbie**" (I've been focusing on learning various languages, *nix command-line, and assembly-level stuff but haven't delved very far into git tutorials available -- read a little only at this point).

From the grin directory, when I type:
```$ git branch --list```

The output shows only the "master" branch.

When I type:
```$ git tag```

there seems to be a number of choices to checkout, but "milestone" is not among them.

**How does one access milestone/anything?**   Apologies for my ignorance (but I have to start from where I'm at to increase my understanding).

**Perhaps I should close or modify the issue because today I did get the bleeding-edge version (3.0.0-alpha.1) to build but I encountered the same problem that caused me wanting to migrate to version 2.1 (the headers are updating but the chain stopped updating blocks at  chain height 403597.**

**Comment by Sviluppo718** [comment]
I checked-out the v2.1.1 tag and the wallet built without errors now.  I've been following the project for several months but haven't yet bought or successfully mined any Grin so I figured I had nothing to lose by running
```$ ./grin clean``` .

The blocks are now caught up with the headers again and I'm going to close the issue.  **If I actually owned some Grin would I have lost them by running the clean command?** (I didn't see clean explained in the documentation directory -- I found it using the '--help' option at the command-line and assumed it was equivalent to the resync command from most Bitcoin-derived coins).

I'm closing this issue.
